src/context/cdhashmap.h

```diff
@@ -355,14 +355,27 @@ class CDHashMap : public ContextObj {
 
     if(i == d_map.end()) {// create new object
       Element* obj = new(true) Element(d_context, this, k, d);
-      d_map[k] = obj;
+      d_map.insert(std::make_pair(k, obj));
       return true;
     } else {
       (*i).second->set(d);
       return false;
     }
   }
 
+  // Use this for pointer data d allocated in context memory at this
+  // level.  THIS IS HIGHLY EXPERIMENTAL.  It seems to work if ALL
+  // your data objects are allocated from context memory.
+  void insertDataFromContextMemory(const Key& k, const Data& d) {
```

Comment:
Martin is about to merge his floating-point work. We decided that it would be best to submit changes touching non-fp parts of the solver as separate PRs. This is one of those PRs.

Reply:
FWIW : I don't recall explictly using this.

Reply:
I really don't think we should add this.  In the early days of CVC4 I remember using functions that looked like this that ended up being buggy.

Reply:
@martin-cs could you do a quick grep on your current code base to check whether you are using it or not? If not, I can remove it from the PR.

Reply:
Here's what I think happened: the original PR from Morgan had this in it because it was in the code base at the time.  We since removed it because it wasn't being used and its correctness is suspect, so probably Martin isn't using it, it was just leftover.  So should be fine to remove!

Reply:
I did a grep in the floating-point-symfpu branch. insertDataFromContextMemory is not used, so it's safe to remove.

Reply:
I support Clark's account.

Reply:
I removed the method.